spec/services/provider_authorisation_spec.rb

```diff
@@ -98,4 +98,89 @@
+    let(:training_provider) { create(:provider) }
+    let(:ratifying_provider) { create(:provider) }
+    let(:accredited_provider) { ratifying_provider }
+    let(:ratifying_permissions) do
+      create(
+        :provider_relationship_permissions,
+        :accredited_body_permissions,
+        ratifying_provider: ratifying_provider,
+        training_provider: training_provider,
+      )
+    end
+    let(:training_permissions) do
+      create(
+        :provider_relationship_permissions,
+        ratifying_provider: ratifying_provider,
+        training_provider: training_provider,
+      )
+    end
+
+    subject(:can_view_safeguarding_information) do
+      described_class.new(actor: provider_user)
+        .can_view_safeguarding_information?(course: course)
+    end
+
+    # rubocop:disable RSpec/NestedGroups
+    context 'when a user is permitted to view safeguarding info for a training provider' do
+      let(:provider_user) { create(:provider_user, providers: [training_provider]) }
+
+      before do
+        provider_user.provider_permissions
+          .find_by(provider: training_provider)
+          .update!(view_safeguarding_information: true)
+      end
+
+      context 'the course is self-ratified' do
+        let(:accredited_provider) { nil }
+
+        it { is_expected.to be true }
+      end
+
+      context 'the course training provider is not permitted, the course accredited provider grants permission' do
+        before { ratifying_permissions.update!(view_safeguarding_information: true) }
+
+        it { is_expected.to be false }
+      end
+
+      context 'the course accredited provider is not permitted, the course training provider grants permission' do
+        before { training_permissions.update!(view_safeguarding_information: true) }
+
+        it { is_expected.to be true }
+      end
+    end
+
+    context 'when a user is permitted to view safeguarding info for the accredited provider' do
+      let(:provider_user) { create(:provider_user, providers: [ratifying_provider]) }
+
+      before do
+        provider_user.provider_permissions
+          .find_by(provider: ratifying_provider)
+          .update!(view_safeguarding_information: true)
+      end
+
+      context 'the course training provider is not permitted, the course accredited provider grants permission' do
+        before { ratifying_permissions.update!(view_safeguarding_information: true) }
+
+        it { is_expected.to be true }
+      end
+
+      context 'the course accredited provider is not permitted, the course training provider grants permission' do
+        before { training_permissions.update!(view_safeguarding_information: true) }
+
+        it { is_expected.to be false }
+      end
+    end
+    # rubocop:enable RSpec/NestedGroups
+
+    context 'when a user is not permitted to view safeguarding info' do
+      let(:provider_user) { create(:provider_user, providers: [training_provider]) }
```

Comment:
Not significant in context, but will the models `ratifying_permissions` and `training_permission` exist here if we don't cause them to be evaluated by calling up the `let`s?

Reply:
No they won't be created, I'd erred on the side of minimal setup here thinking that the test proves we short-circuit on the user permission. I suppose if the logic were reversed in the service method this test would still pass so perhaps it's a bit weak as it stands.

Reply:
Updated here https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/2112/files#diff-1449a26228004ba137dfa4a212a442fcR200-R203 to create and enable these permissions.